### PR TITLE
Fixes issue with projected A(k,w)

### PR DESCRIPTION
**Pull request by harrisonlabollita**
The feature to calculate a projected spectral function is fixed in the ``spaghettis`` method of the ``SumkDFTTools`` class.

### CODE REVIEW

File: python/triqs_dft_tools/sumk_dft_tools.py [hunk 1]
```diff
@@ -831,7 +831,7 @@ def spaghettis(self, broadening=None, plot_shift=0.0, plot_range=None, ishell=No
             subgrp=self.bands_data, things_to_read=things_to_read)
         if not value_read:
             return value_read
-        if ishell:
+        if ishell is not None:
```
Comment (the-hampel) [inline python/triqs_dft_tools/sumk_dft_tools.py:834]: Should we maybe also check if `ishell` is actually a valid integer in the range allowed?

File: python/triqs_dft_tools/sumk_dft_tools.py [hunk 2]
```diff
@@ -897,7 +897,7 @@ def spaghettis(self, broadening=None, plot_shift=0.0, plot_range=None, ishell=No
 
                 for ish in range(self.shells[ishell]['dim']):
                     for sp in spn:
-                        Akw[sp][ish, ik] = -gf.data[numpy.where((mesh > om_minplot)&(mesh < om_maxplot))].imag.trace(axis1=1, axis2=2)/numpy.pi
+                        Akw[sp][ish, ik] = -G_loc[sp].data[numpy.where((mesh > om_minplot)&(mesh < om_maxplot)),ish,ish].imag/numpy.pi
```
Comment (the-hampel) [inline python/triqs_dft_tools/sumk_dft_tools.py:901]: Thanks for this fix. How did this ever work before? This is supposed to give an orbital resolved spectral function for `ishell`, where `ish` is an orbital index? If `ish` is an orbital index, can we name it `iorb` or something?

Comment (harrisonlabollita) [reply python/triqs_dft_tools/sumk_dft_tools.py:901]: In 3.0.x, it was correct (the trace is not needed here). This new version takes advantage of using ``numpy.where`` to quickly slice the mesh window. Yes, ``ish`` is an orbital index. I think ``ish`` is used throughout the code base as the looping index over a ``shell`` dimension, but happy to change to ``iorb``.

Comment (the-hampel) [reply python/triqs_dft_tools/sumk_dft_tools.py:901]: Okay we keep it as it is. This keeps the behavior and fixes the problem! We should maybe consider at some point specifying that the option `ishell` also returns an orbital resolved Akw.

**Review by the-hampel — APPROVED** [review]

**Pull request merged**